zebra-state/src/service/non_finalized_state.rs

```diff
@@ -235,6 +237,10 @@ impl NonFinalizedState {
             self.insert(side_chain);
         }
 
+        // Remove all invalidated_blocks at or below the finalized height
+        self.invalidated_blocks
+            .retain(|height, _blocks| *height > best_chain_root.height);
```

Comment:
If you add a handle to the db as a parameter on `reconsider_block()`, this should leave invalidated blocks until they are below the best chain root height.

```suggestion
            .retain(|height, _blocks| *height >= best_chain_root.height);
```